internal/auth/options.go

```diff
@@ -326,13 +330,22 @@ func AssignProvider(opts *Options) func(*Authenticator) error {
 	}
 }
 
+// SetProvider is a function that takes an Options struct and assigns the
+// appropriate provider to the proxy.
+func SetProvider(provider providers.Provider) func(*Authenticator) error {
+	return func(a *Authenticator) error {
+		a.provider = provider
+		return nil
+	}
+}
+
 // AssignStatsdClient is function that takes in an Options struct and assigns a statsd client
 // to the proxy and provider.
 func AssignStatsdClient(opts *Options) func(*Authenticator) error {
 	return func(proxy *Authenticator) error {
 		logger := log.NewLogEntry()
 
-		StatsdClient, err := newStatsdClient(opts.StatsdHost, opts.StatsdPort)
+		StatsdClient, err := NewStatsdClient(opts.StatsdHost, opts.StatsdPort)
```

Comment:
Instead of creating a new client, can we pass in the old one?

Create a new func:
```
SetStatsdClient(statsdClient *statsd.Client) func ....
```

and do this logic there?

Reply:
I like this idea, though I also like the pattern of just passing in a relevant set of options (Would you pass the previously created statsdclient through the `NewAuthenticatorMux` function for example?) and having it create/assign the necessary components (e.g. the statsdclient). 

I think however creating and storing the statsdclient in the opts struct might be a strange direction. I don't feel particularly strongly regardless - just some general thoughts.